Assets/AppCenter/Editor/AppCenterSettingsContext.cs

```diff
@@ -3,12 +3,35 @@
 
 using UnityEngine;
 using UnityEditor;
+using System.IO;
+using System;
 
 public class AppCenterSettingsContext : ScriptableObject
 {
-    public const string AppCenterPath = "Assets";
-    private const string SettingsPath = AppCenterPath + "/AppCenter/AppCenterSettings.asset";
-    private const string AdvancedSettingsPath = AppCenterPath + "/AppCenter/AppCenterSettingsAdvanced.asset";
+    private static readonly string SettingsPath = AppCenterPath + "/AppCenterSettings.asset";
+    private static readonly string AdvancedSettingsPath = AppCenterPath + "/AppCenterSettingsAdvanced.asset";
+    private static AppCenterSettingsContext Instance;
+
+    public static AppCenterSettingsContext GetInstance()
+    {
+        if (Instance == null)
+        {
+            Instance = new AppCenterSettingsContext();
+        }
+        return Instance;
+    }
+
+    private AppCenterSettingsContext()
```

Comment:
![image](https://user-images.githubusercontent.com/34677812/69864843-e4c6cf80-12b0-11ea-99dc-0748136c0b22.png)
https://forum.unity.com/threads/unityexception-scriptableobject-ctor-is-not-allowed-to-be-called-from-a-scriptableobject.466687/

Reply:
Fixed.

Reply:
Reworked